ocp_resources/node_health_check.py

```diff
@@ -8,3 +8,73 @@ class NodeHealthCheck(Resource):
     """
 
     api_group = Resource.ApiGroup.REMEDIATION_MEDIK8S_IO
+    self_node_remediation_api_group = Resource.ApiGroup.SELF_NODE_REMEDIATION_MEDIK8S_IO
+
+    def __init__(
+        self,
+        name=None,
+        namespace=None,
+        client=None,
+        node_role="worker",
+        min_unhealthy=1,
+        unhealthy_duration="300s",
+        escalating_remediation=False,
+        yaml_file=None,
+        **kwargs,
+    ):
+        """
+            Args
+             name (str): name of the node health check object
+             namespace (str): namespace of node health check
+             node_role (str): role of the node. example: worker, control-plane
+             min_unhealthy (int): Remediation is allowed if at least "MinHealthy" nodes selected by "selector"
+             unhealthy_duration (str): time after which remediation starts
+             escalating_remediation (list): EscalatingRemediations contain a list of ordered remediation templates with
+        a timeout. example: [{'remediationTemplate': {'apiVersion': 'self-node-remediation.medik8s.io/v1alpha1', 'kind':
+        'SelfNodeRemediationTemplate', 'namespace': 'openshift-operators',
+        'name': 'self-node-remediation-resource-deletion-template'}, 'order': 1, 'timeout': '70s'}]
+
+        """
+        super().__init__(
+            name=name,
+            **kwargs,
+        )
+        self.node_role = node_role
+        self.min_unhealthy = min_unhealthy
+        self.unhealthy_duration = unhealthy_duration
+        self.escalating_remediation = escalating_remediation
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
+            self.res.setdefault("spec", {})
+            if self.escalating_remediation:
+                self.res["spec"]["escalatingRemediations"] = self.escalating_remediation
+            if not self.escalating_remediation:
+                # Self Node Remediation is the default remediation for now.
+                self.res["spec"]["remediationTemplate"] = {
+                    "apiVersion": f"{self.self_node_remediation_api_group}/{self.api_version}.split('/')[1]",
+                    "kind": "SelfNodeRemediationTemplate",
+                    "name": "self-node-remediation-resource-deletion-template",
+                    "namespace": self.namespace,
+                }
```

Comment:
do not hard-code anything here; should be provided by the user

Reply:
ack